src/test/scala/org/graphframes/lib/ConnectedComponentsSuite.scala

```diff
@@ -215,18 +215,31 @@ class ConnectedComponentsSuite extends SparkFunSuite with GraphFrameTestSparkCon
   }
 
   test("intermediate storage level") {
-    val friends = Graphs.friends
-    val expected = Set(Set("a", "b", "c", "d", "e", "f"), Set("g"))
-
-    val cc = friends.connectedComponents
-    assert(cc.getIntermediateStorageLevel === StorageLevel.MEMORY_AND_DISK)
-
-    for (storageLevel <- Seq(StorageLevel.DISK_ONLY, StorageLevel.MEMORY_ONLY, StorageLevel.NONE)) {
-      // TODO: it is not trivial to confirm the actual storage level used
-      val components = cc
-        .setIntermediateStorageLevel(storageLevel)
-        .run()
-      assertComponents(components, expected)
+    // disabling adaptive query execution helps assertComponents
```

Comment:
The need for this change here looks like a correctness bug in Spark 3.5.0. I created a PR with a attempted fix here: https://github.com/apache/spark/pull/43188